Mage.Sets/src/mage/cards/i/InspiringStatuary.java

```diff
@@ -21,12 +22,15 @@ public final class InspiringStatuary extends CardImpl {
 
     static {
         filter.add(Predicates.not(CardType.ARTIFACT.getPredicate()));
+        filter.add(Predicates.not(CardType.LAND.getPredicate()));
+        filter.add(Predicates.not(new AbilityPredicate(ImproviseAbility.class))); // So there are not redundant copies being added to each card
     }
 
     public InspiringStatuary(UUID ownerId, CardSetInfo setInfo) {
         super(ownerId, setInfo, new CardType[]{CardType.ARTIFACT}, "{3}");
 
         // Non-artifact spells you cast have improvise.
+        // TODO: Add to the bottom of the card, not the top.
```

Comment:
I think the improvise text on the card should be added a the bottom (between the cards original abilities and the improvise hint). But I'm not sure how.